core/src/main/scala/ai/h2o/sparkling/backend/utils/RestCommunication.scala

```diff
@@ -227,6 +231,59 @@ trait RestCommunication extends Logging with RestEncodingUtils {
 
   private def urlToString(url: URL) = s"${url.getHost}:${url.getPort}"
 
+  private def openUrlConnection(url: URL, conf: H2OConf): HttpURLConnection = {
+    val connection = url.openConnection()
+    if (connection.isInstanceOf[HttpsURLConnection]) {
+      val secureConnection = connection.asInstanceOf[HttpsURLConnection]
+      if (!conf.verifySslCertificates) {
+        disableCertificateVerification(secureConnection)
+      } else if (conf.autoFlowSsl) {
+        setSelfSignedCertificateVerification(secureConnection, conf)
+      }
+
+      // Enable all hosts since the certificate located in the distributed jks file could be created
+      // on a different server.
+      secureConnection.setHostnameVerifier(new HostnameVerifier {
```

Comment:
Side note: H2O on Hadoop (h2odriver) has an option to use a per-host JKS entry, via the hostnameAsJksAlias option.

Basically, you can have a JKS that has certificates for each server and they will be looked-up using the hostname as an alias.